lib/kanta/translations/messages/finders/list_messages.ex

```diff
@@ -7,44 +7,77 @@ defmodule Kanta.Translations.Messages.Finders.ListMessages do
     module: Kanta.Translations.Message,
     binding: :message
 
-  alias Kanta.Translations.{PluralTranslation, SingularTranslation}
+  alias Kanta.Translations.PluralTranslations.Finders.ListPluralTranslations
+  alias Kanta.Translations.SingularTranslations.Finders.ListSingularTranslations
 
   @available_filters ~w(domain_id context_id)
 
   def find(params \\ []) do
     base()
-    |> not_translated_query(params[:filter])
     |> filter_query(Map.take(params[:filter] || %{}, @available_filters))
-    |> search_query(params[:search])
+    |> not_translated_query(params[:filter])
+    |> search_subquery(params[:filter], params[:search])
     |> preload_resources(params[:preloads] || [])
     |> paginate(String.to_integer(params[:page] || "1"), params[:per_page])
   end
 
   defp not_translated_query(query, %{"locale_id" => locale_id, "not_translated" => "true"}) do
     singular_messages_query =
-      from(m in query,
-        join: st in SingularTranslation,
-        on: st.message_id == m.id,
-        where: st.locale_id == ^locale_id,
-        where:
-          (is_nil(st.translated_text) or st.translated_text == "") and
-            (is_nil(st.original_text) or st.original_text == "")
+      query
+      |> with_join(:singular_translations, %{"locale_id" => locale_id})
+      |> where(
+        [singular_translation: st],
+        (is_nil(st.translated_text) or
+           st.translated_text == "") and
+          (is_nil(st.original_text) or
+             st.original_text == "")
```

Comment:
This could be moved to a macro:

```elixir
# WARN: macros have to be defined before use
defmacrop null_or_empty(field) do
    quote do
        fragment("(? = '') IS NOT FALSE", unquote(field))
    end
end
```

Ref: https://stackoverflow.com/questions/23766084/best-way-to-check-for-empty-or-null-value